addons/sale_project/models/sale_order.py

```diff
@@ -88,56 +105,88 @@ def _action_confirm(self):
+        else:  # 1 or 0 tasks -> form view
+            action['views'] = [(form_view_id, 'form')]
+            action['res_id'] = self.tasks_ids.id
         # filter on the task of the current SO
         action.setdefault('context', {})
-        action['context'].update({'search_default_sale_order_id': self.id})
+        # set default project
+        default_project_id = default_line.project_id.id or self.project_id.id or (len(self.project_ids) > 0 and self.project_ids[0].id)
+
+        action['context'].update({
+            'search_default_sale_order_id': self.id,
+            'default_sale_order_id': self.id,
+            'default_sale_line_id': default_line.id,
+            'default_partner_id': self.partner_id.id,
+            'default_project_id': default_project_id,
+            'default_user_ids': [self.env.uid],
+        })
+        return action
+
+    def action_create_project(self):
+        self.ensure_one()
+        if not self.order_line:
+            return {'type': 'ir.actions.act_window_close'}
+
+        sorted_line = self.order_line.sorted(key=lambda sol: sol.sequence)
+        default_sale_line = next(sol for sol in sorted_line if sol.product_id.detailed_type == 'service')
+        action = self.env["ir.actions.actions"]._for_xml_id("project.open_create_project")
+        action['context'] = {
+            'default_sale_order_id': self.id,
+            'default_sale_line_id': default_sale_line.id,
+            'default_partner_id': self.partner_id.id,
+            'default_user_ids': [self.env.uid],
+            'default_allow_billable': 1,
+            'hide_allow_billable': True,
+            'default_company_id': self.company_id.id
+        }
+        if default_sale_line.product_id.service_policy == 'delivered_milestones':
+            action['context'].update({'generate_milestone': True})
         return action
 
     def action_view_project_ids(self):
         self.ensure_one()
-        view_form_id = self.env.ref('project.edit_project').id
-        view_kanban_id = self.env.ref('project.view_project_kanban').id
+        if not self.order_line:
+            return {'type': 'ir.actions.act_window_close'}
+
+        sorted_line = self.order_line.sorted('sequence')
+        default_sale_line = next(sol for sol in sorted_line if sol.product_id.detailed_type == 'service')
         action = {
             'type': 'ir.actions.act_window',
-            'domain': [('id', 'in', self.project_ids.ids)],
-            'view_mode': 'kanban,form',
             'name': _('Projects'),
+            'domain': ['|', ('sale_order_id', '=', self.id), ('id', 'in', self.project_ids.ids)],
             'res_model': 'project.project',
+            'views': [(False, 'kanban'), (False, 'tree'), (False, 'form')],
+            'view_mode': 'kanban,tree,form',
+            'context': {
+                **self.env.context,
```

Comment:
```suggestion
                **self._context,
```
is equivalent, but not required